Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,648 @@
 #define FACECONVERTER_H
 
 #include "CarveHeaders.h"
-
 #include <memory>
-
 #include "ConverterBase.h"
-
 #include "CurveConverter.h"
 #include "PlacementConverter.h"
 #include "GeomUtils.h"
 #include "GeometryInputData.h"
 #include "PlacementConverter.h"
-
 #include "BlueFramework/Core/Diagnostics/log.h"
 
-
 namespace OpenInfraPlatform {
 	namespace Core {
 		namespace IfcGeometryConverter {
 			class ItemData;
 
+			/*! \brief Converter functionality for \c IfcSurfaces's subtypes.
+			*
+			* This class includes converter functions for
+			* \c IfcBoundedSurface and its subtypes,
+			* \c IfcElementarySurface and its subtypes and
+			* \c IfcSweptSurface.
+			*
+			* \param IfcEntityTypesT The IFC version templates
+			*/
+
 			template <
 				class IfcEntityTypesT
 			>
-			class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
+				class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
 			{
 			public:
+
+				//! Constructor
 				FaceConverterT(std::shared_ptr<GeometrySettings> geomSettings,
 					std::shared_ptr<UnitConverter<IfcEntityTypesT>> unitConverter,
 					std::shared_ptr<PlacementConverterT<IfcEntityTypesT>> pc,
 					std::shared_ptr<CurveConverterT<IfcEntityTypesT>> cc)
-					: 
+					:
 					ConverterBaseT<IfcEntityTypesT>(geomSettings, unitConverter),
 					placementConverter(pc),
 					curveConverter(cc)
 				{
 				}
 
+				//! Virtual destructor
 				virtual ~FaceConverterT()
 				{
 				}
 
-				void convertIfcSurface(const std::shared_ptr<typename IfcEntityTypesT::IfcSurface>& surface,
+				double lengthFactor = UnitConvert()->getLengthInMeterFactor();
+
+				/*! \brief Converts \c IfcSurface to a according to subtype.
+					\param		surface	\c IfcSurface entity to be interpreted.
+					\param		pos
+					\param		polylineData
+					\note		The \c IfcSurface subtypes are \c IfcBoundedSurface,  \c IfcElementarySurface and  \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcSurface>& surface,
 					const carve::math::Matrix& pos,
-					std::shared_ptr<carve::input::PolylineSetData>& polyline_data)
-				{
-					double length_factor = UnitConvert()->getLengthInMeterFactor();
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {

```

Comment:
```suggestion
					std::shared_ptr<carve::input::PolylineSetData>& polylineData
				) const throw(...)
				{
```
Please apply this to all functions.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/9ea30708f673b500fde8a81ee4c7baf6ef37c375